### PR TITLE
Modify tkn version output to hide components version if they are not installed

### PR DESCRIPTION
**Pull request by Divyansh42**
# Changes

Modified tkn version output to hide Triggers and Dashboard version if they are not installed.

Closes: #1143 

Signed-off-by: Divyansh42 <diagrawa@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes
 ```release-note
`tkn version` output is modified to hide components version if they are not installed
```

### CODE REVIEW

**Comment by Divyansh42** [comment]
Do we need to hide the `pipeline version` if pipeline is not installed?

**Review by vinamra28 — COMMENTED** [review]
can you add some `unit-tests` for this?

**Comment by danielhelfand** [comment]
> Do we need to hide the `pipeline version` if pipeline is not installed?

I would say yes. Only show versions if the component is installed.

**Comment by vdemeester** [comment]
> > Do we need to hide the `pipeline version` if pipeline is not installed?
> 
> I would say yes. Only show versions if the component is installed.

Maybe a warning ? (to say "well, `tkn` won't work well without any tekton instance :stuck_out_tongue_closed_eyes:")

**Review by vdemeester — APPROVED** [review]
In any case, I am fine getting this in and continue discussing what to do or change for Pipeline in a follow-up/issue.

**Comment by Divyansh42** [comment]
@nikhil-thomas @danielhelfand @piyush-garg @vdemeester I just saw PR #1092 got merged, and it is contradicting with this PR, I want to remove the version of tekton component if that is not installed whereas #1092 we are displaying the version `unknown` with a message/suggestion to check another namespace.

*Output after PR #1092*
```
$ tkn version
Client version: dev
Pipeline version: unknown, pipeline controller may be installed in another namespace please use tkn version -n {namespace}
Triggers version: unknown, triggers controller may be installed in another namespace please use tkn version -n {namespace}
Dashboard version: unknown, dashboard controller may be installed in another namespace please use tkn version -n {namespace}
```
*This PR's output after rebasing with the master branch*
```
$ tkn version
Client version: dev
Pipeline version: unknown, pipeline controller may be installed in another namespace please use tkn version -n {namespace}
```
Please let me know your thoughts on it. :slightly_smiling_face:

**Comment by nikhil-thomas** [comment]
> @nikhil-thomas @danielhelfand @piyush-garg @vdemeester I just saw PR #1092 got merged, and it is contradicting with this PR, I want to remove the version of tekton component if that is not installed whereas #1092 we are displaying the version `unknown` with a message/suggestion to check another namespace.
> 
> _Output after PR #1092_
> 
> ```
> $ tkn version
> Client version: dev
> Pipeline version: unknown, pipeline controller may be installed in another namespace please use tkn version -n {namespace}
> Triggers version: unknown, triggers controller may be installed in another namespace please use tkn version -n {namespace}
> Dashboard version: unknown, dashboard controller may be installed in another namespace please use tkn version -n {namespace}
> ```
> 
> _This PR's output after rebasing with the master branch_
> 
> ```
> $ tkn version
> Client version: dev
> Pipeline version: unknown, pipeline controller may be installed in another namespace please use tkn version -n {namespace}
> ```
> 
> Please let me know your thoughts on it.

we really need to read the version from a clusterScoped resource. :smile: 

@Divyansh42 so does that mean that after your rebase
`Pipeline version: unknown, pipeline controller may be installed in another namespace please use tkn version -n {namespace}
`
will show up only for pipelines. Will the other 2 version lines be hidden it they are not installed.
if yes, i think it should be alright.

**Comment by pradeepitm12** [comment]
> Pipeline version: unknown, pipeline controller may be installed in another namespace please use tkn version -n {namespace}
will show up only for pipelines. Will the other 2 version lines be hidden it they are not installed.

this seems good to me.

**Comment by vdemeester** [comment]
> _his PR's output after rebasing with the master branch_
> 
> ```
> $ tkn version
> Client version: dev
> Pipeline version: unknown, pipeline controller may be installed in another namespace please use tkn version -n {namespace}
> ```

This sounds good to me. I don't think it contradicts anything. One PR did allow to check on different namespaces, this one removes triggers and dashboard version if we can't get it — it's orthogonal. Any way, I would go with the above ^^

**Comment by Divyansh42** [comment]
Okay, thanks :+1: Will update the PR :slightly_smiling_face:

**Comment by Divyansh42** [comment]
/cc @piyush-garg @vdemeester @piyush-garg @danielhelfand 
Added unit tests, PR can be reviewed now :slightly_smiling_face:

**Comment by vdemeester** [comment]
Still looking good
/cc @savitaashture @pradeepitm12 @danielhelfand

File: pkg/cmd/version/testdata/TestGetVersions-deployment_only_with_pipeline_installed.golden [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Client version: dev
+Pipeline version: unknown, pipeline controller may be installed in another namespace please use tkn version -n {namespace}
```
Comment (danielhelfand) [inline pkg/cmd/version/testdata/TestGetVersions-deployment_only_with_pipeline_installed.golden:2]: I think we should tweak this error message, but I can put in a follow up pr. It shouldn't delay this pr.

**Review by danielhelfand — APPROVED** [review]

**Review by vinamra28 — APPROVED** [review]

**Review by piyush-garg — APPROVED** [review]
/lgtm

**Pull request merged**